### PR TITLE
[2102] Get the number of threads from destination dataObjInp

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This is the same as the one for 4-2-stable except for:

1. The rodsLog() statements have been replaced by the logger.
2. I ran clang format.  (I did not keep the spaces to tabs conversion.)

The tests passed.

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Add pound when ready.

File: s3/s3_operations.cpp [hunk 1]
```diff
@@ -487,9 +523,12 @@ namespace irods_s3 {
...
         }
 
-        get_number_of_threads_data_size_and_opr_type(_ctx, number_of_threads, data_size, oprType);
+		ret = get_number_of_threads_data_size_and_opr_type(_ctx, number_of_threads, data_size, oprType);
+		if (!ret.ok()) {
+			return std::make_tuple(PASS(ret), data.dstream_ptr, data.s3_transport_ptr);
+		}
 
-        logger::debug("{}:{} ({}) [[{}]] oprType set to {}", __FILE__, __LINE__, __FUNCTION__, thread_id, oprType);
+		logger::debug("{}:{} ({}) [[{}]] oprType set to {}", __FILE__, __LINE__, __FUNCTION__, thread_id, oprType);
```
Comment (alanking) [inline s3/s3_operations.cpp:531]: See some tabs down here in this file

Comment (JustinKyleJames) [reply s3/s3_operations.cpp]: I fixed that and a couple of other places.

There are still a couple of code segments that have tabs from previous pull requests/formats.  I can fix that later but don't want to confuse people by making those changes here.

**Comment by JustinKyleJames** [comment]
I fixed the tab/space issue and pounded it.

**Review by alanking — APPROVED** [review]

**Pull request merged**